### PR TITLE
kwargs is renamed to additional_arguments

### LINKED ISSUE #4458: unfortunate use of kwarg 'kwargs' results in some arguments not having effect
In a couple of places (e.g. `rasa.train` and `rasa.core.train`) we have a keyword argument called `kwargs`.
This is a confusing name as `**kwargs` is conventionally used to capture 0 or more additional, optional keyword arguments. 
These function signatures should be changed, either by using a different name for this variable (`additional_arguments`) or by adapting them to use `**kwargs`.

This also causes the argument `dump_stories` of `rasa.train` to have no effect at all. (See https://forum.rasa.com/t/how-can-i-use-dump-stories-from-script/15433)

Update:
Another place where we use `kwargs` as a name for a dict instead of a container for optional keyword arguments is https://github.com/RasaHQ/rasa/blob/bd83681222f6ad60997df40964c51bf79858df2e/rasa/test.py#L37

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
-  fix #4458 

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/test.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def test_core(
             "Rasa model and provide it via the '--model' argument."
         )
 
-    use_e2e = kwargs["e2e"] if "e2e" in kwargs else False
+    use_e2e = additional_arguments.get("e2e", False)
```
Comment (wochinge) [inline rasa/test.py]: 🚀

File: rasa/core/train.py [hunk 2]
```diff
@@ -26,10 +26,9 @@
     output_path: Text,
     interpreter: Optional["NaturalLanguageInterpreter"] = None,
     endpoints: "AvailableEndpoints" = None,
-    dump_stories: bool = False,
```
Comment (wochinge) [inline rasa/core/train.py]: I don't really like the dictionary approach in general because we are loosing the typing information and have magic strings to access the parameter. How about creating a class with types and named things? We could also do this in a separate PR? So revert the `dump_stories` part and we create an issue for the type / handling of `additional_arguments`?

Comment (alwx) [reply rasa/core/train.py]: 👍

File: changelog/4458.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Rename ``kwargs`` to ``additional_arguments`` in ``rasa.train`` and ``rasa.core.train`` to make the name of the
```
Comment (wochinge) [inline changelog/4458.bugfix.rst]: Let's make it `misc` changelog entry since it doesn't really affect users
